src/components/LogRow/CollapsedRow/index.tsx

```diff
@@ -3,47 +3,83 @@ import styled from "@emotion/styled";
 import Button from "@leafygreen-ui/button";
 import { palette } from "@leafygreen-ui/palette";
 import { Overline } from "@leafygreen-ui/typography";
-import { ListRowProps } from "react-virtualized";
 import Icon from "components/Icon";
 import { size } from "constants/tokens";
+import { OverlineType } from "types/leafygreen";
+import { BaseRowProps } from "../types";
 
-const { gray } = palette;
+const { gray, black } = palette;
 
-interface CollapsedRowProps extends ListRowProps {
-  numCollapsed: number;
-}
+const SKIP_NUMBER = 5;
 
-const CollapsedRow = forwardRef<any, CollapsedRowProps>((props, ref) => {
-  const { index, numCollapsed, ...rest } = props;
+const CollapsedRow = forwardRef<any, BaseRowProps>((props, ref) => {
+  const { data, listRowProps } = props;
+  const { processedLines, expandLines } = data;
+  const { index } = listRowProps;
+
+  const line = processedLines[index] as number[];
+  const numCollapsed = line.length;
+  const start = line[0];
+  const end = line[line.length - 1];
 
   const lineText =
     numCollapsed !== 1 ? `${numCollapsed} lines skipped` : "1 line skipped";
+  const disableExpandFive = 2 * SKIP_NUMBER > numCollapsed;
+
+  const expandFive = () => {
+    expandLines([
+      [start, start + (SKIP_NUMBER - 1)],
+      [end - (SKIP_NUMBER - 1), end],
+    ]);
+  };
+
+  const expandAll = () => {
+    expandLines([[start, end]]);
+  };
```

Comment:
We can considerably speed this up by taking advantage of some of the concurrency features built into react 18
https://reactjs.org/docs/hooks-reference.html#usetransition
```suggestion
  const [isPending, startTransition] = useTransition();
  const expandFive = () => {
  startTransition(() => {
    expandLines([
      [start, start + (SKIP_NUMBER - 1)],
      [end - (SKIP_NUMBER - 1), end],
    ]);
    });
  };

  const expandAll = () => {
   startTransition(() => {
    expandLines([[start, end]]);
    });
  };
```